community/kernel/src/main/java/org/neo4j/kernel/GraphDatabaseQueryService.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2002-2016 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel;
+
+import java.net.URL;
+
+import org.neo4j.graphdb.DependencyResolver;
+import org.neo4j.graphdb.Label;
+import org.neo4j.graphdb.Node;
+import org.neo4j.graphdb.Relationship;
+import org.neo4j.graphdb.Transaction;
+import org.neo4j.graphdb.security.URLAccessValidationError;
+
+/*
+ * This is a trimmed down version of GraphDatabaseService and GraphDatabaseAPI, limited to a subset of functions needed
+ * by implementations of QueryExecutionEngine.
+ */
+public interface GraphDatabaseQueryService
```

Comment:
Yes. We were easily able to get rid of most access to GDS, but a few places looked like they required a little more thought, so we postponed them. Once this version is stable, we want to do a second round of cleanup, by looking for (and removing if possible) all access to getGraphDatabaseService(). It is not trivial mainly because cypher calls other modules that are themselves expecting GDS. So while we want the cypher code to not depend on GDS, it is hard to enforce this separation in the libraries that it calls deeper down.